artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalImpl.java

```diff
@@ -2245,6 +2379,155 @@ private synchronized JournalLoadInformation load(final LoaderCallback loadManage
       }
    }
 
+
+   @Override
+   public void processBackupCleanup() {
+      if (journalRetentionFolder != null && (journalRetentionMaxFiles > 0 || journalRetentionPeriod > 0)) {
+
+         FilenameFilter fnf = new FilenameFilter() {
+            @Override
+            public boolean accept(final File file, final String name) {
+               return name != null && name.endsWith("." + filesRepository.getFileExtension());
+            }
+         };
+
+
+         if (journalRetentionPeriod > 0) {
+            String[] fileNames = journalRetentionFolder.list(fnf);
+            Arrays.sort(fileNames);
+
+            GregorianCalendar calendar = this.calendarThreadLocal.get();
+            calendar.setTimeInMillis(System.currentTimeMillis() - journalRetentionUnit.toMillis(journalRetentionPeriod));
+            long timeCutOf = calendar.getTimeInMillis();
+
+            for (String fileName : fileNames) {
+               long timeOnFile = getDatePortionMillis(fileName);
+               if (timeOnFile < timeCutOf) {
+                  logger.debug("File " + fileName + " is too old and should go");
+                  File fileToRemove = new File(journalRetentionFolder, fileName);
+                  if (!fileToRemove.delete()) {
+                     logger.debug("Could not remove " + fileToRemove);
+                  }
+               }
+            }
+         }
+
+         if (journalRetentionMaxFiles > 0) {
+            String[] fileNames = journalRetentionFolder.list(fnf);
+            Arrays.sort(fileNames);
+
+            if (fileNames.length > journalRetentionMaxFiles) {
+               int toRemove = fileNames.length - journalRetentionMaxFiles;
+
+               for (String file : fileNames) {
+                  logger.debug("Removing " + file);
+                  File fileToRemove = new File(journalRetentionFolder, file);
+                  fileToRemove.delete();
+                  toRemove--;
+                  if (toRemove <= 0) {
+                     break;
+                  }
+               }
+            }
+         }
+
+
+      }
+   }
+
+   @Override
+   public void processBackup() {
+      if (journalRetentionFolder != null) {
+         ArrayList<JournalFile> filesToMove;
+         synchronized (historyPendingFiles) {
+            filesToMove = new ArrayList<>(historyPendingFiles.size());
+            filesToMove.addAll(historyPendingFiles);
+            historyPendingFiles.clear();
+         }
+
+
+         Calendar calendar = calendarThreadLocal.get();
```

Comment:
Why in some places using gregorian and her just plain calendar

Reply:
It doesn't make a difference really? the threadLocal is still a GregorianCalendar.

Reply:
because of the change i was thinking there was maybe more to it.  I guess just consistency so in future people not double thinking like i just did why the difference